src/vs/workbench/browser/parts/editor/multiEditorTabsControl.ts

```diff
@@ -515,16 +515,15 @@ export class MultiEditorTabsControl extends EditorTabsControl {
 		// redraw of tabs.
 
 		const activeEditorChanged = this.didActiveEditorChange();
-		const oldActiveTabLabel = this.activeTabLabel;
-		const oldTabLabelsLength = this.tabLabels.length;
+		const oldTabLabels = this.tabLabels;
 		this.computeTabLabels();
 
 		// Redraw and update in these cases
 		let didChange = false;
 		if (
-			activeEditorChanged ||													// active editor changed
-			oldTabLabelsLength !== this.tabLabels.length ||							// number of tabs changed
-			!this.equalsEditorInputLabel(oldActiveTabLabel, this.activeTabLabel)	// active editor label changed
+			activeEditorChanged ||																				// active editor changed
+			oldTabLabels.length !== this.tabLabels.length ||													// number of tabs changed
+			oldTabLabels.some((label, index) => !this.equalsEditorInputLabel(label, this.tabLabels.at(index))) 	// editor labels changed
```

Comment:
@benibenj unfortunately we have to compare all labels, we actually had a bug where rendering was broken in tabs in this case:
* first tab is a preview editor
* second tab is active and non-preview
* an editor opens as `inactive: true` where the active editor will remain the same
* this causes the first preview editor to close and a new preview editor to open after the active one
* the overall number of tabs is the same though